pkg/operation/botanist/component/kubeapiserver/kube_apiserver.go

```diff
@@ -307,8 +310,10 @@ func (k *kubeAPIServer) Deploy(ctx context.Context) error {
 		return err
 	}
 
-	if err := k.reconcileSecretUserKubeconfig(ctx, secretStaticToken, secretBasicAuth); err != nil {
-		return err
+	if enableAdminKubeconfig {
+		if err := k.reconcileSecretUserKubeconfig(ctx, secretStaticToken, secretBasicAuth); err != nil {
+			return err
+		}
```

Comment:
When `EnableAdminKubeconfig` is `false` then this secret should be deleted. I would implement this in https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/component/kubeapiserver/secrets.go#L131 and not here (than above `if` condition should also be moved to this function).

Reply:
I have tested it on local. The secret is getting deleted when EnableAdminKubeconfig is set to false. I thought it will good we don't call the `reconcileSecretUserKubeconfig` function entirely when `enableAdminKubeconfig ` si set to false. WDYT ?

Reply:
Ah, nice, this is happening because we call `Cleanup` of secrets manager. Yeah, all good then, nothing to be done. Keep it as is!